src/EFCore.Relational/Query/QuerySqlGenerator.cs

```diff
@@ -790,6 +775,56 @@ protected override Expression VisitIn(InExpression inExpression)
         protected virtual string GetOperator(SqlBinaryExpression binaryExpression)
             => _operatorMap[binaryExpression.OperatorType];
 
+        /// <summary>
+        ///     Returns a bool value indicating if the inner SQL expression required to be put inside parenthesis
+        ///     when generating SQL for outer SQL expression.
+        /// </summary>
+        /// <param name="outerExpression">The outer expression which provides context in which SQL is being generated.</param>
+        /// <param name="innerExpression">The inner expression which may need to be put inside parenthesis.</param>
+        /// <returns>A bool value indicating that parenthesis is required or not. </returns>
+        protected virtual bool RequiresParenthesis(SqlExpression outerExpression, SqlExpression innerExpression)
```

Comment:
nit: parentheses is the plural of parenthesis (there are two)

```suggestion
        protected virtual bool RequiresParentheses(SqlExpression outerExpression, SqlExpression innerExpression)
```

Reply:
I considered it and left it for API review. Will update name.